src/react/boosts/payments/BoostPayForm.tsx

```diff
@@ -109,6 +159,15 @@ export class BoostPayForm extends React.Component<BoostPayFormProps, BoostPayFor
     }
   };
 
+  private postTransaction = async (result: any, txHash: TxHash) => {
```

Comment:
this is fine for now, but we actually need to refactor this so that the mutation happens as soon as the transaction is submitted, rather than when it is confirmed. that way the payment is still tracked even if the user navigates away from the page

Reply:
refactor the whole transaction button / TwoStepEthTransaction thing, but I don't actually think it's a ton of work

Reply:
```
        handleTransactionHash: (txHash: TxHash) => {
          this.props.updateTransactionStatusModalsState({
            isWaitingTransactionModalOpen: false,
            isTransactionProgressModalOpen: true,
          });
        },
```

actually no refactor is necessary, just need to use the handleTransactionHash to send the mutation instead